pkg/image/api/helper.go

```diff
@@ -81,10 +81,20 @@ func ParseDockerImageReference(spec string) (DockerImageReference, error) {
 		ref.Name = repoParts[0]
 		ref.Tag = tag
 		ref.ID = id
-		return ref, nil
+		break
 	default:
 		return ref, fmt.Errorf("the docker pull spec %q must be two or three segments separated by slashes", spec)
 	}
+
+	if len(ref.Namespace) == 0 {
```

Comment:
We had the defaulting for `registry/name` but not for `name`. What code do we have that expects no namespace in a pull spec?